### ISSUE TITLE
properly document the package filename format

### ISSUE THREAD

**Issue by umlaeute** [pure-data/deken#109]
there is no proper documentation about the filename format (apart from the source, both the filename-generator and the filename parser).

e.g. many packages uploaded by @electrickery have a slightly non-conformant `version` section, which might make the various parsers fail:

     cyclone-0.2beta1(Linux-armv7l-32)(Sources)-externals.tgz

should read:

    cyclone-v0.2beta1-(Linux-armv7l-32)(Sources)-externals.tgz

**Comment by umlaeute** [comment]
so the format is really something like:

     LIBNAME[-vVER-]{(ARCH)}-externals.EXT

with:
- `LIBNAME` library name (e.g. `zexy`)
- `VER` the version (e.g. `2.3`)
- `ARCH` the architecture specifier:
  - `Sources` for source-package
  - or `OS-MACH-BIT`, with
    - `OS`: Operating System (e.g. `Linux`)
    - `MACH`: machine architecture (e.g. `x86_64`)
    - `BIT` some number of bits (e.g. `64`)
- `EXT` the archive extension (either `zip` or `tgz`)

in plain english: 

> the library-name, followed by an optional version string (starting with `-v` and ending with `-`), followed by zero or more architecture specifications (each surrounded by `(`parantheses`)`), and terminated by `-externals` (followed by a filename extension).

examples:

    adaptive-v0.0.extended-(Linux-i386-32)(Linux-amd64-64)-externals.tar.gz
    adaptive-v0.0.extended-(Sources)-externals.tar.gz
    freeverb~(Darwin-i386-32)(Darwin-x86_64-32)(Sources)-externals.zip
    iemnet-v0.2.1~git20151118--externals.zip

the `VER` field is optional (see 2nd example), though not using it is discouraged (to be able to distinguish between multiple versions of the same library; if the package has no official release version, make something up, e.g. based on the current date `0.0.20151224`

the `ARCH` field shall be omitted for *architecture independent* libraries (e.g. abstraction-only libraries).
note the double dash `--externals`, if such a package has a `VER` field!

**Comment by electrickery** [comment]
@umlaeute Fixed the uploaded packages.

**Comment by umlaeute** [comment]
here is the actual regex we are using:

     (.*/)?(.+?)(-v(.+)-)?((\([^\)]+\))+|-)*-externals\.([a-z.]*)

with the following matching groups:
- `#0` anything before the path (always empty and *ignored*)
- `#1` = path to filename (*ignored*)
- `#2` = libname
- `#3` = version string with decoration (*ignored*)
- `#4` = version
- `#5` = archs
- `#6` = last arch in archs (*ignored*)
- `#7` = extension
- `#8` anything after the extension (should be empty and is *ignored*)

if somebody can suggest a better one (esp. one that gets rid of the double-dash for libraries without archs) i'd be thankful (e.g. currently i can't remember what that `|-` in the archs-groups is supposed to catch.
